components/match2/wikis/fifa/match_group_input_custom.lua

```diff
@@ -0,0 +1,624 @@
+	local casters = {}
+	for key, name in Table.iter.pairsByPrefix(match, 'caster') do
+		table.insert(casters, CustomMatchGroupInput._getCasterInformation(
+			name,
+			match[key .. 'flag'],
+			match[key .. 'name']
+		))
+	end
+
+	match.extradata = {
+		casters = Table.isNotEmpty(casters) and Json.stringify(casters) or nil,
+		hassubmatches = tostring(Logic.readBool(match.hasSubmatches)),
+	}
+end
+
+function CustomMatchGroupInput._adjustData(match)
+	CustomMatchGroupInput._opponentInput(match)
+
+	for _, _, mapIndex in Table.iter.pairsByPrefix(match, 'map') do
+		CustomMatchGroupInput._mapInput(match, mapIndex)
+	end
+
+	local scores = Array.map(Array.range(1, MAX_NUM_OPPONENTS), function(opponentIndex)
+		local score = CustomMatchGroupInput._computeOpponentMatchScore(match, opponentIndex, true)
+		match['opponent' .. opponentIndex].score = score
+		if Logic.isNumeric(score) then
+			match['opponent' .. opponentIndex].status = SCORE_STATUS
+		end
+		return score
+	end)
+
+	CustomMatchGroupInput._winnerProcessing(match, scores, true)
+
+	CustomMatchGroupInput._setPlacements(match)
+end
+
+function CustomMatchGroupInput._winnerProcessing(obj, scores, isMatch)
+	walkoverProcessing.walkover(obj, scores)
+
+	if isMatch and obj.resulttype == DEFAULT_WIN_RESULTTYPE then
+		walkoverProcessing.applyMatchWalkoverToOpponents(obj)
+		return
+	end
+
+	if obj.winner == 'draw' then
+		obj.winner = 0
+		obj.resulttype = 'draw'
+		obj.finished = true
+	end
+
+	obj.finished = CustomMatchGroupInput._checkFinished(obj)
+
+	obj.winner = tonumber(obj.winner)
+
+	if obj.finished and not obj.winner then
+		obj.winner = CustomMatchGroupInput._determineWinnerIfMissing(scores)
+	end
+end
+
+function CustomMatchGroupInput._computeOpponentMatchScore(match, opponentIndex)
+	if not match['opponent' .. opponentIndex] then
+		return NO_SCORE
+	end
+
+	-- if valid manual input return that
+	local score = match['opponent' .. opponentIndex].score
+	if Logic.isNumeric(score) then
+		return tonumber(score)
+	elseif Table.includes(ALLOWED_STATUSES, (score or ''):upper()) then
+		return score:upper()
+	end
+
+	if not match.map1 or not match.map1.winner then
+		return NO_SCORE
+	end
+
+	local sumScore = 0
+
+	-- if submatches count submatch/map wins
+	if Logic.readBool(match.hasSubmatches) then
+		for _, map in Table.iter.pairsByPrefix(match, 'map') do
+			if map.winner == opponentIndex then
+				sumScore = sumScore + 1
+			end
+		end
+
+		return sumScore
+	end
+
+	-- if won in penalty shoot out return score of that shoot out
+	for _, map in Table.iter.pairsByPrefix(match, 'map') do
+		if Logic.readBool(map.penalty) then
+			return map.scores[opponentIndex]
+		end
+	end
+
+	-- sum up scores
+	for _, map in Table.iter.pairsByPrefix(match, 'map') do
+		local mapScore = map.scores[opponentIndex]
+		if mapScore ~= NO_SCORE then
+			sumScore = sumScore + mapScore
+		end
+	end
+
+	return sumScore
+end
+
+function CustomMatchGroupInput._checkFinished(obj)
```

Comment:
```suggestion
function CustomMatchGroupInput._isFinished(obj)
```

Reply:
Forgot to set resolve this one (change made at 179 and additionally at 122 too )